File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
class Solution {
    private int len, now, opStart, opEnd;
    
    private int toSeconds(String time) {
        String[] parts = time.split(":");
        int m = Integer.parseInt(parts[0]);
        int s = Integer.parseInt(parts[1]);
        
        return m * 60 + s;
    }
    
    private String toTime(int seconds) {
        int m = seconds / 60;
        int s = seconds % 60;
        
        return String.format("%02d:%02d", m, s);
    }
    
    private void skipOp() {
        if (opStart <= now && now < opEnd) {
            now = opEnd;
        }
    }
    
    private void runCommand(String command) {
        switch(command) {
            case "prev":
                now = Math.max(0, now - 10);
                break;
            case "next":
                now = Math.min(len, now + 10);
                break;
            default: return;
        }
    }
    
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        this.len = toSeconds(video_len);
        this.now = toSeconds(pos);
        this.opStart = toSeconds(op_start);
        this.opEnd = toSeconds(op_end);
        
        skipOp();
        for (int i = 0; i < commands.length; i++) {
            runCommand(commands[i]);
            skipOp();
        }
        
        return toTime(now);
    }
}

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.kt
class Solution {
    private var len = 0
    private var now = 0
    private var opStart = 0
    private var opEnd = 0
    
    private fun toSeconds(time: String): Int = time.split(":").let { it[0].toInt() * 60 + it[1].toInt() }
    
    private fun toTime(seconds: Int): String {
        val m = seconds / 60
        val s = seconds % 60
        
        return "%02d:%02d".format(m, s)
    }
    
    private fun skipOp() {
        if (now in opStart until opEnd) {
            now = opEnd
        }
    }
    
    private fun runCommand(command: String) {
        when (command) {
            "prev" -> now = maxOf(0, now - 10)
            "next" -> now = minOf(len, now + 10)
        }
    }
    
    fun solution(video_len: String, pos: String, op_start: String, op_end: String, commands: Array<String>): String {
        len = toSeconds(video_len)
        now = toSeconds(pos)
        opStart = toSeconds(op_start)
        opEnd = toSeconds(op_end)

        skipOp()
        for (command in commands) {
            runCommand(command)
            skipOp()
        }

        return toTime(now)
    }
}